src/lib/Hydra/Controller/API.pm

```diff
@@ -88,7 +88,7 @@ sub jobsetToHash {
         triggertime => $jobset->triggertime,
         fetcherrormsg => $jobset->fetcherrormsg,
         errortime => $jobset->errortime,
-        haserrormsg => $jobset->errormsg eq "" ? JSON::false : JSON::true
+        haserrormsg => defined($jobset->errormsg) ? ($jobset->errormsg eq "" ? JSON::false : JSON::true) : JSON::false
```

Comment:
I think it's clearer to collapse the nested condition into a single condition like:
```suggestion
        haserrormsg => defined($jobset->errormsg) && $jobset->errormsg ne "" ? JSON::true : JSON::false
```